File: _core/packages/python/openproblems/src/openproblems/utils/strip_margin.py
def strip_margin(text: str, symbol: str = "\\|") -> str:
  """
  Strip margin from a string

  Args:
      text (str): A character vector.
      symbol (str): The margin symbol to strip.
  
  Returns:
      str: A character vector with the margin stripped.

  Example:

    ```python
    strip_margin("
      |hello_world:
      |  this_is: "a yaml"
      |")
    ```
  """

  import re
  return re.sub("(^|\n)[ \t]*" + symbol, "\\1", text)

File: _core/packages/r/openproblems.utils/R/strip_margin.R
#' Strip margin from a string
#'
#' @param text A character vector.
#' @param symbol The margin symbol to strip.
#' @return A character vector with the margin stripped.
#'
#' @export
#' @examples
#' strip_margin(
#'   "hello_world:
#'   |  this_is: \"a yaml\""
#' )
strip_margin <- function(text, symbol = "\\|") {
  gsub(paste0("(^|\n)[ \t]*", symbol), "\\1", text)
}
